src/NuGet.Indexing/Extraction/DocumentCreator.cs

```diff
@@ -261,6 +262,61 @@ private static void AddRequiresLicenseAcceptance(Document document, IDictionary<
             }
         }
 
+        private static void AddPackageTypes(Document document, IDictionary<string, string> package)
+        {
+            string flattenedPackageTypes;
+            StringBuilder searchIndex = new StringBuilder();
+            if (package.TryGetValue("flattenedPackageTypes", out flattenedPackageTypes))
```

Comment:
This `IDictionary<string, string>` is just an intermediate data structure, right? Why do that `flattenedPackageTypes` here? In fact, why have a flattened format at all? This is a new field everywhere. I think we should have a consistent model unless we have a very compelling reason.